build.sh

```diff
@@ -1,14 +1,38 @@
 #!/bin/sh
+set -e
+
+if [ -z "$CC" ]; then
+    CC=gcc
+fi
+
 rm NAR
 rm src/RuleTable.c
-set -e
-Str=`ls src/*.c src/NetworkNAR/*.c | xargs`
-echo $Str
+sources=`ls src/*.c src/NetworkNAR/*.c | xargs`
+echo $sources
 echo "Compilation started:"
-BaseFlags="-mfpmath=sse -msse2 -pthread -lpthread -D_POSIX_C_SOURCE=199506L -pedantic -std=c99 -g3 -O3 $Str -lm -oNAR"
+BaseFlags="-pthread -lpthread -D_POSIX_C_SOURCE=199506L -pedantic -std=c99 -g3 -O3 $sources -lm -oNAR"
 NoWarn="-Wno-tautological-compare -Wno-dollar-in-identifier-extension -Wno-unused-parameter -Wno-unused-variable"
-gcc -DSTAGE=1 -Wall -Wextra -Wformat-security $NoWarn $BaseFlags
+
+# check if we are on Windows, use SSE2 then
+if [ "$OSTYPE" == "cygwin" ]; then
```

Comment:
A double equals (`==`) is bash-specific and does not work in /bin/sh

On the other hand, bash defines it as "does the same thing as a single equals", so I have no earthly clue why it exists, and strongly discourage anyone from ever using the double equals version under any circumstances ever. Even if you use it in a bash script instead of a /bin/sh script, you are training yourself to see it as normal, and then find yourself accidentally using it in /bin/sh scripts.